src/slices/posts.slice.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type Posts = {
+  loaded: boolean;
+  hasError: boolean;
+  items: Post[];
+};
+
+const initialState: Posts = {
+  loaded: true,
+  hasError: false,
+  items: [],
+};
+
+export const fetchPosts = createAsyncThunk<Post[], number>(
+  'posts/fetchPosts',
+  async (userId: number) => {
+    const posts = await getUserPosts(userId);
+
+    return posts;
+  },
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    resetPost: state => {
+      state.items = [];
```

Comment:
Consider also resetting 'loaded' and 'hasError' to their initial values in the 'resetPost' reducer to ensure the state is fully reset.